src/app/components/search/search.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss']
+})
+
+export class SearchComponent implements OnInit {
+  searchForm = new FormGroup({
+    title: new FormControl(),
+    description: new FormControl(),
+    street: new FormControl(),
+    state: new FormControl(),
+    company: new FormControl(),
+    city: new FormControl(),
+    type: new FormControl(),
+    purchase_plan: new FormControl(),
+    garages: new FormControl(),
+    bathrooms: new FormControl(),
+    bedrooms: new FormControl(),
+    lot_size: new FormControl(),
+    price_min: new FormControl(),
+    price_max: new FormControl(),
+  })
+
+  constructor(
+    private spinner: NgxSpinnerService,
+    private router: Router,
+  ) { }
+
+  ngOnInit() { }
+
+  search({ value }) {
+    this.spinner.show();
+    let urlValues = Object.keys(value).reduce((acc, key) => value[key] ? acc + key + '=' + value[key] + '&' : acc, '')
+    urlValues = urlValues.substring(0, urlValues.length - 1);
+    this.router.navigate(['/properties'], { queryParams: { q: urlValues } })
```

Comment:
@JamesMudidi , we also need a semicolon here. Other than that
your feature looks good to me.. great job working on this feature and making improvements even though you were under the weather